libp2p/security/secio/pb/spipe.proto

```diff
@@ -0,0 +1,16 @@
+syntax = "proto2";
+
+package spipe.pb;
+
+message Propose {
+  optional bytes rand = 1;
+  optional bytes public_key = 2;
+  optional string exchanges = 3;
```

Comment:
Should this be `curves`? Or otherwise should it's type be `Exchange`?

Reply:
i was just following the `proto` file here: https://github.com/libp2p/go-libp2p-secio/blob/master/pb/spipe.proto

i do agree `curves` may be more descriptive but in terms of aligning to the de facto reference impl, i'll keep it like this for now